src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -235,6 +241,47 @@ public void writeLogTo(XMLOutput out) throws IOException {
         @Inject
         private transient GitHubHookRegisterProblemMonitor monitor;
 
+        @Inject
+        private transient SCMTrigger.DescriptorImpl scmTrigger;
+
+        private transient int maximumThreads = Integer.MIN_VALUE;
+
+        private static ThreadFactory threadFactory() {
+            return new NamingThreadFactory(Executors.defaultThreadFactory(), "GitHubPushTrigger");
+        }
+
+        public DescriptorImpl() {
+            checkThreadPoolSize();
+        }
+
+        /**
+         * Update the {@link java.util.concurrent.ExecutorService} instance.
+         */
+        /*package*/
+        synchronized void checkThreadPoolSize() {
+            if (scmTrigger != null) {
+                int count = scmTrigger.getPollingThreadCount();
+                if (maximumThreads != count) {
+                    maximumThreads = count;
+                    try {
+                        Field getQueue = SCMTrigger.DescriptorImpl.class.getDeclaredField("queue");
+                        getQueue.setAccessible(true);
```

Comment:
of course... but after a more careful analysis I decided against that and rather just mirror the pool size... which sadly means you have two pools... but this does point that perhaps we should have an explicit pool for polling that plugins can use rather than the current status where each push notifier has to roll their own... with hillarious consequences